core/src/main/java/org/ehcache/core/internal/util/ClassLoading.java

```diff
@@ -36,7 +36,7 @@ public static ClassLoader getDefaultClassLoader() {
   }
 
   public static <T> ServiceLoader<T> libraryServiceLoaderFor(Class<T> serviceType) {
-    return ServiceLoader.load(serviceType, ClassLoading.class.getClassLoader());
+    return ServiceLoader.load(serviceType, getDefaultClassLoader());
```

Comment:
The method is called `libraryServiceLoader(...)`. My understanding is that the intent is to not look for thing outside the scope of the classloader that loaded the Ehcache classes.  Services are a facility for dynamically binding together the various internal modules of Ehcache to form an implementation of the API.  I don't think we want to discovering implementations of services discovered in other a wider classloader context.

Reply:
That kinda defeats the purpose of using the Java service framework.

Reply:
No it doesn't. Our use of the Java service framework is intended to be an internal detail, not a supported piece of public API.